app/controllers/teachers_controller.rb

```diff
@@ -0,0 +1,9 @@
+class TeachersController < ApplicationController
+  def index
+    application = Application.find_by(user_id: current_user.id);
+    if application.application_status != "accepted"
+      flash[:danger] = "Access denied because user application status does not equal 'accepted'"
```

Comment:
I think we should come up with another error message - this is what the user will see and it may seem strange to them. Maybe something like "Your application must first be accepted in order to access this page."